fair_mango/metrics/superset.py

```diff
@@ -0,0 +1,318 @@
+    EqualisedOddsDifference,
+    EqualisedOddsRatio,
+    EqualOpportunityDifference,
+    EqualOpportunityRatio,
+    FalsePositiveRateDifference,
+    FalsePositiveRateRatio,
+    PerformanceMetric,
+    SelectionRate,
+)
+
+
+def super_set_fairness_metrics(
+    metric: (
+        type[DemographicParityDifference]
+        | type[DemographicParityRatio]
+        | type[DisparateImpactDifference]
+        | type[DisparateImpactRatio]
+        | type[EqualOpportunityDifference]
+        | type[EqualOpportunityRatio]
+        | type[EqualisedOddsDifference]
+        | type[EqualisedOddsRatio]
+        | type[FalsePositiveRateDifference]
+        | type[FalsePositiveRateRatio]
+    ),
+    data: Dataset | pd.DataFrame,
+    sensitive: Sequence[str] | None = None,
+    real_target: Sequence[str] | None = None,
+    predicted_target: Sequence[str] | None = None,
+    positive_target: Sequence[int | float | str | bool] | None = None,
+) -> list:
+    """Calculate fairness metrics for different subsets of sensitive
+    attributes. Ex:
+    [gender, race] → (gender), (race), (gender, race)
+
+    Parameters
+    ----------
+    metric : type[DemographicParityDifference]  |  type[DemographicParityRatio]
+    |  type[DisparateImpactDifference]  |  type[DisparateImpactRatio]  |
+    type[EqualOpportunityDifference]  |  type[EqualOpportunityRatio]  |
+    type[EqualisedOddsDifference]  |  type[EqualisedOddsRatio]  |
+    type[FalsePositiveRateDifference]  |  type[FalsePositiveRateRatio]
+        The fairness metric class to be used for evaluation
+    data : Dataset | pd.DataFrame
+        The dataset containing the data to be evaluated. If a DataFrame object
+        is passed, it should contain attributes `sensitive`, `real_target`,
+        `predicted_target`, and `positive_target`.
+    sensitive : Sequence[str], optional
+        A Sequence of sensitive attributes (Ex: gender, race...), by default []
+    real_target : Sequence[str] | None, optional
+        A Sequence of column names of actual labels for target variables,
+        by default None
+    predicted_target : Sequence[str] | None, optional
+        A Sequence of column names of predicted labels for target variables,
+        by default None
+    positive_target : Sequence[int  |  float  |  str  |  bool] | None, optional
+        A Sequence of the positive labels corresponding to the provided
+        targets, by default None
+
+    Returns
+    -------
+    list
+        list
+        A list of dictionaries, each containing the sensitive attributes
+        considered and their corresponding fairness metric result.
+
+    Raises
+    ------
+    AttributeError
+        if data is a pandas dataframe and 'sensitive' parameter is not provided.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({
+    ...         'gender': ['male', 'male', 'male', 'female', 'female'],
+    ...         'race': ['white', 'black', 'black', 'white', 'white'],
+    ...         'real': [1,1,0,0,1],
```

Comment:
use comprehensive names like "real_churn" so that we don't take the "real" for a key named after the `real_target` parameter of `super_set_fairness_metrics`

Reply:
Okay, I will make sure to do this change in the documentation branch too.